src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -1156,4 +1158,131 @@ function ($a) {
             ->end()
         ;
     }
+
+    private function addHttpClientSection(ArrayNodeDefinition $rootNode)
+    {
+        $subNode = $rootNode
+            ->children()
+                ->arrayNode('http_client')
+                    ->info('Http Client configuration')
+                    ->canBeEnabled()
+                    ->fixXmlConfig('http_client')
+                    ->fixXmlConfig('api_client')
+                    ->children()
+                        ->integerNode('max_host_connections')
+                            ->info('The maximum number of connections to a single host.')
+                            ->defaultValue(6)
+                        ->end()
+                        ->scalarNode('no_proxy')
+                            ->info('A comma separated list of hosts that do not require a proxy to be reached.')
+                            ->defaultNull()
+                        ->end();
+        $subNode = $this->addHttpClientOptionsSection($subNode);
```

Comment:
Can't we use `append()` here? 

And why is there both `http_client` and `http_clients`? If the purpose is to have a default maybe add a `default: true` key for `http_clients`, but warn if this is used more then once 🤔 or `default_http_client` (al though a bit of context).